clustering/__init__.py

```diff
@@ -4,53 +4,25 @@
 from sklearn.decomposition import PCA
 from sklearn.cluster import KMeans
 
-
-def find_incident(labels):
-    res = set()
-    for i in range(len(labels)):
-        if labels[i].startswith('i_'):
-            res.add(i)
-    return res
-
-
-# numpy array construction
-def construct_np_allin(allin, m, n, incident_labels):
-    np_allin = []
-    for i in range(m):
-        tmp = []
-        count = 0
-        for j in range(n):
-            if allin[i][j] == True:
-                tmp.append(1)
-            elif allin[i][j] == False:
-                tmp.append(0)
-            elif allin[i][j] != 'None':
-                tmp.append(1)
-            else:
-                tmp.append(0)
-            if j in incident_labels and allin[i][j] == 1:
-                count += 1
-        np_allin.append(tmp+[count])
-    return np_allin
-
+def _create_incidents_matrix(dataframe):
+    false_values = [False, 'False', None, 'None', 0]
+    # Map values to binary
+    transformed = np.isin(dataframe.values, false_values, invert=True)
+    # Add counts of incident labels set to 1
+    counts = np.count_nonzero(dataframe.loc[:, dataframe.columns.str.startswith('i_')], axis=1)
+    counts = counts.reshape(counts.shape[0],-1)
+    # Add the counts as a last column for each entry
+    return np.hstack((transformed, counts))
 
 def cluster(pd_rules):
-    # global incident_labels, m, n, pd_allin
-    np_rules = pd_rules.as_matrix()
-    labels = list(pd_rules.columns.values)
-    allin = np_rules[:, 3:]
-    allin_labels = labels[3:]
-    incident_labels = find_incident(allin_labels)
-    m, n = allin.shape
-    np_allin = construct_np_allin(allin, m, n, incident_labels)
-    np_allin = np.array(np_allin)
+    np_allin = _create_incidents_matrix(pd_rules)
+
+    # TODO: Fix me
     pd_allin = pd.DataFrame(data=np_allin[:, :-1], columns=allin_labels)
     # Kmeans clustering model pipeline
     min_max_scaler = preprocessing.StandardScaler()
```

Comment:
Why is the data scaled when it's just `True` or `False` values? And why is it scaled twice?